Userland/Applications/Assistant/Providers.cpp

```diff
@@ -62,24 +62,31 @@ void URLResult::activate() const
     Desktop::Launcher::open(URL::create_with_url_or_path(title()));
 }
 
+AppProvider::AppProvider()
```

Comment:
I'm surprised that this is a major bottleneck. Have you measured it?

Reply:
Yes, the majority of the performance improvement in this PR is from this change.

This is because the AppFile constructor opens and parses a config file. On my machine there are 68 AppFiles returned. Prior to this change, this was happening on every key press.

Here is a profile with the `AppProvider` change reverted. I've expanded the function I've optimized:
![assistant_profile_appprovider_reverted](https://user-images.githubusercontent.com/2817754/231670286-0d616584-1153-4596-ae8c-113b1579309d.png)

Reply:
Fine then, I was confused by it not being shown in the previous profile.